src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -249,11 +284,10 @@ public SafeDeleteContext(SafeFreeCredentials credential, bool isServer, bool rem
             }
             catch(Exception ex)
             {
-                if (gotCredRef)
-                {
-                    _credential.DangerousRelease();
-                }
                 Debug.Write("Exception Caught. - " + ex);
+                Debug.Assert((null != credential), "Null credential in SafeDeleteContext");
+                credential.DangerousRelease();
```

Comment:
It was called in the base ctor. Do you want the AddRef/ReleaseRef to be moved from the base SafeDeleteContext class to SafeDeleteSslContext and SafeDeleteNegoContext?